Signal/src/ViewControllers/ConversationView/MessagesViewController.m

```diff
@@ -561,6 +561,13 @@ - (void)viewWillAppear:(BOOL)animated
     // or on another device.
     [self hideInputIfNeeded];
 
+    // We need to `beginLongLivedReadTransaction` before we update our
+    // mapping in order to jump to the most recent commit.
+    [self.uiDatabaseConnection beginLongLivedReadTransaction];
```

Comment:
Right - it's the `messageMappings` that is inconsistent, and before we update it below, we want to make sure that uiDatabaseConnection has latest.  You'll notice we do this before `toggleObservers`, where we begin to observe database modified events.  

I don't have a repro for this transient issue, but I think this is the right approach.